src/confidential.rs

```diff
@@ -95,6 +95,17 @@ impl GetInfo<ConfidentialAssetInfo> for Asset {
 	}
 }
 
+impl GetInfo<ConfidentialAssetInfo> for AssetId {
```

Comment:
This seems like it's just a useful impl but isn't used anywhere yet, is that right?

Reply:
Good question. No, it is actually used. 

In `tx.rs`:
```
#[derive(Clone, PartialEq, Eq, Debug, Deserialize, Serialize)]
pub struct PeginDataInfo {
	pub outpoint: String,
	pub value: u64,
	pub asset: ConfidentialAssetInfo,
	pub genesis_hash: bitcoin::BlockHash,
	pub claim_script: HexBytes,
	pub mainchain_tx_hex: HexBytes,
	pub mainchain_tx: Option<hal::tx::TransactionInfo>,
	pub merkle_proof: HexBytes,
	pub referenced_block: bitcoin::BlockHash,
}

impl<'tx> GetInfo<PeginDataInfo> for PeginData<'tx> {
	fn get_info(&self, network: Network) -> PeginDataInfo {
		PeginDataInfo {
			outpoint: self.outpoint.to_string(),
			value: self.value,
			asset: self.asset.get_info(network),
			genesis_hash: self.genesis_hash,
			claim_script: self.claim_script.into(),
			mainchain_tx_hex: self.tx.into(),
			mainchain_tx: match bitcoin::consensus::encode::deserialize::<bitcoin::Transaction>(&self.tx) {
				Ok(tx) => Some(hal::GetInfo::get_info(&tx, BTCNET)),
				Err(_) => None,
			},
			merkle_proof: self.merkle_proof.into(),
			referenced_block: self.referenced_block,
		}
	}
}
```

This previously worked because `PeginData.asset` had type `Asset`. However, n ew versions of the `elements` crate have changed the `PeginData.asset` field to be of type `AssetId`